Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -823,42 +880,50 @@ bool qSlicerSaveDataDialogPrivate::saveNodes()
     QString format = this->format(row);
     qSlicerIOOptions* options = this->options(row);
     vtkMRMLNode* node = vtkMRMLNode::SafeDownCast(this->object(row));
+    vtkMRMLStorageNode* const storageNode = vtkMRMLStorageNode::SafeDownCast(this->object(row));
 
     if (file.fileName().isEmpty())
       {
-      QMessageBox::warning(this, tr("Saving node..."),
-                           tr("Node %1 not saved, file name is empty.").arg(nodeNameItem->text()));
+      if (storageNode)
+        {
+        storageNode->GetUserMessages()->AddMessage({vtkCommand::WarningEvent,
```

Comment:
I think it would make more sense to move this logic so that the save button is not enabled unless the proper fields have been filled out.  Basically add a validation function that is run when any ui elements are edited.

Reply:
I fear that I am not understanding what you are saying.  With the current implementation, 

1. any user messages from the `vtkMRMLStorageNode` nodes that occur _before_ the Save Data Dialog box is displayed will get shown to the user in a new last column of that dialog box.
1. when the Save button is clicked, if there are any additional warnings and errors then the dialog box will get updated to also have the new messages.  

Am I correct that you are suggesting a step between those two?  That is, if any warnings or errors are displayed in that first step, we should gray out the Save button and otherwise make the user do something in order to have the right to Save?

Or am I way off base and the two implemented steps above are not what you were expecting?

Reply:
At today's Slicer Hangout we discussed trying the approach of showing icons for warnings and errors, with a tool tip having the more complete message.